File: ServerSpace/src/main/java/DataLayer/MonedaDB.java
package DataLayer;

import EntityLayer.MonedaEntity;
import Framework.BaseDB;
import Framework.injector;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class MonedaDB extends BaseDB {

    injector InjDB = new injector();

    public ArrayList<MonedaEntity> ObtenerMonedaItems() {
        ArrayList<MonedaEntity> DatoMemoria = new ArrayList<>();
        MonedaEntity en;
        try {
            InjDB.Sp("sp_MonedaObtenerItems");
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new MonedaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public ArrayList<MonedaEntity> ObtenerMonedaItem(int MonedaId) {

        ArrayList<MonedaEntity> DatoMemoria = new ArrayList<>();
        MonedaEntity en;
        try {
            InjDB.Sp("sp_MonedaObtenerItem");
            InjDB.Pmt_Integer("@MonedaId", MonedaId, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new MonedaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

}

File: Space/Space.DataLayer/MonedaDB.cs
using Framework;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.DataLayer
{
    public class MonedaDB : BaseDB
    {

        public List<MonedaEntity> ObtenerMonedaItems()
        {
            try
            {

                List<MonedaEntity> ListaData = new List<MonedaEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_MonedaObtenerItems");
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    MonedaEntity entity = new MonedaEntity();
                    if (FillFrom<MonedaEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                Conexion.CloseConnection();
                return ListaData;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }


        public List<MonedaEntity> ObtenerMonedaItem(Int32 MonedaId)
        {
            try
            {
                List<MonedaEntity> ListaData = new List<MonedaEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_MonedaObtenerItem");
                Conexion.AddParemeter("MonedaId", SqlDbType.Int, 10, ParameterDirection.Input, MonedaId);
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    MonedaEntity entity = new MonedaEntity();
                    if (FillFrom<MonedaEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                return ListaData;

            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }
    }
}
